src/Tuura/Concept/Circuit/Basic.hs

```diff
@@ -35,17 +36,19 @@ data Concept s e a = Concept
                    {
                        initial   :: a -> InitialValue,
                        arcs      :: [([e], e)],
-                       interface :: a -> Interface
+                       interface :: a -> Interface,
+                       invariant :: [[e]]
                    }
```

Comment:
I think this could work very nicely as a definition for `Invariant`: 
```haskell
data Invariant e = NeverAll [e]

never es = invariantConcept [NeverAll es] 
```

That is, at the moment, the only type of invariants we support are `NeverAll` invariants. We could extend this list later.